cmd/mt-whisper-importer-writer/verify.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+
+# this is a tool to verify if the data imported via the whisper importer is
+# actually correct. it uses two inputs, the output of whisper-dump.py and the
+# output of mt-store-cat.
+#
+# pipe the whisper-dump.py output into a file:
+# $> whisper-dump.py storage/whisper/some/id/of/a/metric/1.wsp  > /tmp/whisper_dump
+#
+# pipe the mt-store-cat output into another file:
+# $> mt-store-cat -print-ts -from 1487764800 -to 1487785801 normal id 1.d588ebb28e4d2ca64d828cb1eb6066d4 5h > /tmp/mt_store_dump
```

Comment:
Yes, let me just show a full example... 
* I generate a whisper file with the schema `30s:7d, 1m:60d, 10m:1y`
* I import it with
```
./mt-whisper-importer-writer -verbose=true  -http-endpoint=localhost:1234  -cassandra-addrs=192.168.56.101:9042 -cassandra-keyspace=metrictank -exit-on-error=true -fake-avg-aggregates=true -ttls=7d,60d,1y
./mt-whisper-importer-reader -chunkspans=10min,1h,12h -exit-on-error=true -orgid=1 -write-unfinished-chunks=false -threads=10 -whisper-directory=/tmp/importTest -http-endpoint='http://localhost:1234/chunks' -name-prefix=my.test. -verbose=true
```
* I dump the whisper file with:
```
whisper-dump.py /tmp/importTest/a/b/c/d/e/1.wsp  > /tmp/whisper_dump
```
* I manually split the archives in `/tmp/whisper_dump` into the `3` files `/tmp/whisper_dump1`, `/tmp/whisper_dump2`, `/tmp/whisper_dump3`
* I use `mt-store-cat` to dump the `id` which `mt-whisper-importer-writer` printed during the import (`1.12991776b2f712109766c5a9a284fe93`)
```
cmd/mt-store-cat/mt-store-cat -cassandra-keyspace metrictank -print-ts normal id 1.12991776b2f712109766c5a9a284fe93 7d > /tmp/mt_store_dump1
cmd/mt-store-cat/mt-store-cat -cassandra-keyspace metrictank -from 1457967244 -print-ts normal id 1.12991776b2f712109766c5a9a284fe93_sum_60 60d > /tmp/mt_store_dump2
cmd/mt-store-cat/mt-store-cat -cassandra-keyspace metrictank -from 1457967244 -print-ts normal id 1.12991776b2f712109766c5a9a284fe93_sum_600 1y > /tmp/mt_store_dump3
```
* I use `verify.py` to compare the dumps
```
mst@ubuntu:~/go/src/github.com/raintank/metrictank$ cmd/mt-whisper-importer-writer/verify.py  /tmp/whisper_dump1 /tmp/mt_store_dump1
got 20158 lines from whisper and 2780 lines from mt_store
misses: 17378
correct: 2780
wrong: 0
mst@ubuntu:~/go/src/github.com/raintank/metrictank$ cmd/mt-whisper-importer-writer/verify.py  /tmp/whisper_dump2 /tmp/mt_store_dump2 60
got 86398 lines from whisper and 86395 lines from mt_store
misses: 5
correct: 86393
wrong: 0
mst@ubuntu:~/go/src/github.com/raintank/metrictank$ cmd/mt-whisper-importer-writer/verify.py  /tmp/whisper_dump3 /tmp/mt_store_dump3 600
got 52557 lines from whisper and 51840 lines from mt_store
misses: 720
correct: 51837
wrong: 0
```
* I look at the number of misses for each combination and compare the dumps in `vi` side-by-side. The last `ts` in the `mt-store-cat` dump is always the last one before the `T0` of the last chunk, the number of lines after that `T0` (and including) is equal to the number of misses, which means all the misses come from `-write-unfinished-chunks=false`